src/components/sidenav/sidenav.js

```diff
@@ -278,31 +307,53 @@ function SidenavDirective($mdMedia, $mdUtil, $mdConstant, $mdTheming, $animate,
      * Toggle the SideNav view and attach/detach listeners
      * @param isOpen
      */
-    function updateIsOpen(isOpen) {
+    function updateIsOpen(isOpen, oldValue) {
+      if (skipNextUpdate) {
+        skipNextUpdate = false;
+        return;
+      }
+
       // Support deprecated md-sidenav-focus attribute as fallback
       var focusEl = $mdUtil.findFocusTarget(element) || $mdUtil.findFocusTarget(element,'[md-sidenav-focus]') || element;
       var parent = element.parent();
 
+      // Temporary remove md-closed class (won't affect the view, because it's outside of the viewport).
+      // And then check the visbility due hide attributes. So if the isOpen variable changes
+      // and a hide attribute is active we should revert the digest change
+      if (!skipSidenav) {
+        var wasClosed = element.hasClass('md-closed');
+        element.removeClass('md-closed');
+        if ($mdUtil.isHidden(element, true, false) && !skipSidenav) {
+          element.toggleClass('md-closed', wasClosed);
+          skipNextUpdate = true;
+          scope.isOpen = !!oldValue;
+          return;
+        }
+        element.toggleClass('md-closed', wasClosed);
```

Comment:
This still needs the `return`, but yes this will work, because we check before removing the class :+1:.

-> Updated